tomviz/FileFormatManager.cxx

```diff
@@ -16,87 +19,150 @@ FileFormatManager& FileFormatManager::instance()
 }
 
 template <typename T>
-void registerFactories(const QString& registerFunction,
-                       QMap<QString, T*>& factories)
+QMap<QString, T*> registerFactories(const QString& name,
+                                    const QString& registerFunction)
 {
-  if (!factories.isEmpty()) {
-    return;
-  }
+  QMap<QString, T*> factories;
+
   Python python;
   auto module = python.import("tomviz.io._internal");
 
   if (!module.isValid()) {
     qCritical() << "Failed to import tomviz.io._internal module.";
-    return;
+    return factories;
   }
 
   auto lister = module.findFunction(registerFunction);
   if (!module.isValid()) {
     qCritical()
       << QString("Failed to find tomviz.io._internal.%1").arg(registerFunction);
-    return;
+    return factories;
   }
 
   auto res = lister.call();
   if (!res.isValid()) {
     qCritical() << QString("Error calling %1.").arg(registerFunction);
-    return;
+    return factories;
   }
 
   if (res.isList()) {
     Python::List formatList(res);
+    QList<QPair<QString, QStringList>> cached;
     for (int i = 0; i < formatList.length(); ++i) {
       Python::List formatInfo(formatList[i]);
       if (!formatInfo.isList() || formatInfo.length() != 3) {
-        return;
+        continue;
       }
       QString description = formatInfo[0].toString();
       Python::List extensions_(formatInfo[1]);
       if (!extensions_.isList()) {
-        return;
+        continue;
       }
       Python::Object readerClass = formatInfo[2];
       QStringList extensions;
       for (int j = 0; j < extensions_.length(); ++j) {
         extensions << QString(extensions_[j].toString());
       }
+      cached << QPair<QString, QStringList>(description, extensions);
       T* factory = new T(description, extensions, readerClass);
       foreach (auto extension, extensions) {
         factories[extension] = factory;
       }
     }
+
+    auto settings = pqApplicationCore::instance()->settings();
+    settings->beginWriteArray(name);
+    for (int i = 0; i < cached.size(); ++i) {
+      settings->setArrayIndex(i);
+      settings->setValue("description", cached.at(i).first);
```

Comment:
Prefer [] over at as you already did bounds checking in the loop initialization.